File: src/main/java/com/github/lkqm/paper/core/WebConfig.java
package com.github.lkqm.paper.core;

import com.github.lkqm.paper.util.FileUtils;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;

import javax.annotation.Resource;
import java.io.File;

@Configuration
public class WebConfig extends WebMvcConfigurerAdapter {

    @Resource
    PaperProperties properties;

    @Override
    public void addResourceHandlers(ResourceHandlerRegistry registry) {
        String workDir = properties.getWorkDir();
        FileUtils.makeParent(workDir);
        File dir = new File(workDir);
        String path = dir.getAbsolutePath();
        String url = "/v/**";
        String location = "file:" + path + File.separator + PaperProperties.PROJECTS_DIR_NAME + File.separator;
        registry.addResourceHandler(url).addResourceLocations(location);
        super.addResourceHandlers(registry);
    }
}

File: src/main/java/com/github/lkqm/paper/config/WebConfig.kt
package com.github.lkqm.paper.config

import com.github.lkqm.paper.util.FileUtils
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter
import java.io.File
import javax.annotation.Resource

/**
 * WEB配置：自定义静态资源目录
 */
@Configuration
open class WebConfig : WebMvcConfigurerAdapter() {

    @Resource
    lateinit var properties: PaperProperties

    override fun addResourceHandlers(registry: ResourceHandlerRegistry) {
        FileUtils.makeParent(properties.workDir)
        val dir = File(properties.workDir)
        val location = "file:" + dir.absolutePath + File.separator + PaperProperties.PROJECTS_DIR_NAME + File.separator
        registry.addResourceHandler("/v/**").addResourceLocations(location)
        super.addResourceHandlers(registry)
    }
}